app/src/androidTest/kotlin/at/bitfire/davdroid/network/DnsRecordResolverTest.kt

```diff
@@ -69,8 +70,9 @@ class DnsRecordResolverTest {
         // entries are selected randomly (for load balancing)
         // run 1000 times to get a good distribution
         val counts = IntArray(2)
+        val seededRandom = Random(0) // Seed of 0 ensures 100% deterministic non-failing results
```

Comment:
Does it? The Kotlin RNG may change at any time.

And even if it's deterministic, we need a lot of runs (here: 1000). Do you think it would work to mock a RNG that provides an exact value for each case? The `bestSRVRecord` algorithm is defined in the DNS standard. So we could, if the algorithm calculcates a value of let's say 0…10, return

- a mocked value of for instance 0 for the first case and see whether bestSRVRecord returns the expected result of the first case,
- a mocked value of for instance 5 for the seconds case and see whether bestSRVRecord returns the expected result of the seconds case,
- …

As I image it, it wouldn't be that complex, but allow us to exactly test each case without having multiple runs.

Reply:
Fair. Please take a look at the changes.